version: '3.8'

services: #Servicios que corremos en nuestra red de docker.
  app:
    build:
      context: .  #AQUI VIENEN LOS PROBLEMAS :) ./docker/python
      dockerfile: ./docker/python/Dockerfile #/Dockerfile
    ports:
      - '5000:5000'
    environment:
      - FLASK_ENV=development
    #volumes:
    #  - .:/app
    depends_on:
      - db

  db:
    build:
      context: ./docker/db
      dockerfile: Dockerfile
    environment:
      POSTGRES_DB: microblogging
      POSTGRES_USER: user
      POSTGRES_PASSWORD: password
    volumes:
      - postgres_data:/var/lib/postgresql/data #Mirar persistencia con los volumenes

  pgadmin:
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: "admin@admin.com"
      PGADMIN_DEFAULT_PASSWORD: "root"
    ports:
      - "8080:80"
    depends_on:
      - db
    volumes:
      postgres_data:
      

  minio:
    image: minio/minio
    volumes:
      - minio_data:/data
    ports:
      - "9000:9000"
    environment:
      MINIO_ACCESS_KEY: minioadmin
      MINIO_SECRET_KEY: minioadmin
    command: server /data
  volumes:
    minio_data:
